wasm.md

```diff
@@ -684,12 +688,131 @@ Currently, only one memory may be accessible to a module, and thus the `<memAddr
+
+    rule <k> store { WIDTH EA VAL } => . ... </k>
+         <memAddrs> 0 |-> ADDR </memAddrs>
+         <memInst>
+           <memAddr> ADDR </memAddr>
+           <msize>   SIZE </msize>
+           <mdata>   DATA => #clearRange(DATA, EA, EA +Int WIDTH -Int 1) [EA := VAL ] </mdata>
+           ...
+         </memInst>
+         requires (EA +Int WIDTH /Int 8) <=Int (SIZE *Int #pageSize())
+
+    rule <k> store { WIDTH  EA  _ } => trap ... </k>
+         <memAddrs> 0 |-> ADDR </memAddrs>
+         <memInst>
+           <memAddr> ADDR </memAddr>
+           <msize>   SIZE </msize>
+           ...
+         </memInst>
+         requires (EA +Int WIDTH /Int 8) >Int (SIZE *Int #pageSize())
+
+    syntax StoreOp ::= "store" | "store8" | "store16" | "store32"
+ // -------------------------------------------------------------
+    rule <k> ITYPE . store   EA VAL => store { #numBytes(ITYPE) EA VAL            } ... </k>
+    rule <k> _     . store8  EA VAL => store { 8                EA #wrap(8,  VAL) } ... </k>
+    rule <k> _     . store16 EA VAL => store { 16               EA #wrap(16, VAL) } ... </k>
+    rule <k> i64   . store32 EA VAL => store { 32               EA #wrap(32, VAL) } ... </k>
+```
+
+The assorted load operations take an address of type `i32`.
+The `loadX_sx` operations loads `X` bits from memory, and extend it to the right length for the return value, interpreting the bytes as either signed or unsigned according to `sx`.
+The value is fethced from the "effective address", which is the address given on the stack plus offset.
+
+```k
+    syntax Instr ::= "(" IValType  "." LoadOpM Instr ")" | "(" IValType  "." LoadOpM ")"
+ //                | "(" FValType  "." LoadOpM Instr ")" | "(" FValType  "." LoadOpM ")"
+                   | IValType "." LoadOp Int
+                   | "load" "{" IValType Int Int Signedness"}"
+    syntax LoadOpM ::= LoadOp | LoadOp MemArg
+    syntax Signedness ::= "Signed" | "Unsigned"
+ // -------------------------------------------
+    rule <k> ( ITYPE . LOPM:LoadOpM I:Instr ) => I ~> ( ITYPE . LOPM ) ... </k>
+
+    rule <k> ( ITYPE . LOP:LoadOp              ) => ITYPE . LOP  IDX                          ... </k>
+         <stack> < i32 > IDX : STACK => STACK </stack>
+    rule <k> ( ITYPE . LOP:LoadOp MEMARG:MemArg) => ITYPE . LOP (IDX +Int #getOffset(MEMARG)) ... </k>
+         <stack> < i32 > IDX : STACK => STACK </stack>
+
+    rule <k> load { ITYPE WIDTH EA SIGN } =>
+      < ITYPE > #if SIGN ==K Signed
+                #then #signedWidth(WIDTH, #range(DATA, EA, WIDTH /Int 8))
+                #else #range(DATA, EA, WIDTH /Int 8)
+                #fi ... </k>
+         <memAddrs> 0 |-> ADDR </memAddrs>
+         <memInst>
+           <memAddr> ADDR </memAddr>
+           <msize>   SIZE </msize>
+           <mdata>   DATA </mdata>
+           ...
+         </memInst>
+         requires (EA +Int WIDTH /Int 8) <=Int (SIZE *Int #pageSize())
```

Comment:
De-indent `requires` to make it more visually apparent (usually I line up the `r` in `requires` with the `l` in `rule`).

Reply:
Done